app/gui/view/component-browser/searcher-list-panel/src/column_grid.rs

```diff
@@ -0,0 +1,189 @@
+use ensogl_core::display;
+use ensogl_core::display::object::ObjectOps;
+use ensogl_gui_component::component;
+use ensogl_list_view as list_view;
+use ide_view_component_group as component_group;
+use ide_view_component_group::Layers;
+use ordered_float::OrderedFloat;
+
+
+
+// ==========================
+// === Constants + Colors ===
+// ==========================
+
+const COLUMN_GAP: f32 = 2.0;
+
+const BASE_COLORS: &[color::Rgba] = &[
+    color::Rgba::new(0.527, 0.554, 0.18, 1.0),
+    color::Rgba::new(43.0 / 255.0, 117.0 / 255.0, 239.0 / 255.0, 1.0),
+    color::Rgba::new(62.0 / 255.0, 139.0 / 255.0, 41.0 / 255.0, 1.0),
+    color::Rgba::new(192.0 / 255.0, 71.0 / 255.0, 171.0 / 255.0, 1.0),
+];
+
+const fn get_base_color_for_index(ix: usize) -> color::Rgba {
+    BASE_COLORS[ix % BASE_COLORS.len()]
+}
+
+
+
+// =============
+// === Model ===
+// =============
+
+/// Contains a `AnyModelProvider` with a label. Can be used to populate a `component_group::View`.
+#[derive(Clone, Debug, Default)]
+pub struct LabeledAnyModelProvider {
+    /// Label of the data provided to be used as a header of the list.
+    pub label:   String,
+    /// Content to be used to populate a list.
+    pub content: list_view::entry::AnyModelProvider<component_group::Entry>,
+}
+
+
+
+// =============
+// === Model ===
+// =============
+
+#[derive(Clone, Debug, CloneRef)]
+pub struct Model {
+    app:            Application,
+    display_object: display::object::Instance,
+    content:        Rc<RefCell<Vec<component_group::View>>>,
+    size:           Rc<Cell<Vector2>>,
+}
+
+impl Model {
+    fn new(app: &Application) -> Self {
+        let logger = Logger::new("ColumnGrid");
+        let app = app.clone_ref();
+        let display_object = display::object::Instance::new(&logger);
+
+        Self {
+            app,
+            display_object,
+            content: Rc::new(RefCell::new(vec![])),
+            size: Rc::new(Cell::new(Vector2::default())),
+        }
+    }
+
+    fn set_content(&self, content: &[LabeledAnyModelProvider]) -> Vector2 {
+        let overall_width = crate::WIDTH_INNER - 2.0 * crate::PADDING_INNER;
+        let column_width = (overall_width - 2.0 * COLUMN_GAP) / 3.0;
+        let content = content
+            .iter()
+            .map(|LabeledAnyModelProvider { content, label }| {
+                let view = self.app.new_view::<component_group::View>();
+                view.set_width(column_width);
+                view.set_entries(content);
+                view.set_header(label.as_str());
+                self.display_object.add_child(&view);
+                view
+            })
+            .collect_vec();
+
+        let mut columns = vec![vec![], vec![], vec![]];
+        let mut heights = [-COLUMN_GAP; 3]; // We need to subtract one COLUMN_GAP as we only need
+                                            // (n-1) gaps, but through iteration below we add one
+                                            // gap per item.
+
```

Comment:
1. `3` is magic number again.
2. Please move this comment above the line - you'd use less space in code and it would be easier to read.